trie/triedb/pathdb/layertree.go

```diff
@@ -142,7 +142,32 @@ func (tree *layerTree) cap(root common.Hash, layers int) error {
 			return err
 		}
 		// Replace the entire layer tree with the flat base
-		tree.layers = map[common.Hash]layer{base.rootHash(): base}
+		// tree.layers = map[common.Hash]layer{base.rootHash(): base}
+		// XXX
```

Comment:
Seems geth adopts a different semantics for commit than what we rely on in blockchain.go
I implemented this version which allows uncommitted layers to remain on top instead of flattening everything.